### PR TITLE
Feature | Error space highlight

### PR DESCRIPTION
**Pull request by ThatcherK**
Ticket https://trello.com/c/sXS8wOQ9/241-highlight-blank-space-in-attribute-codes
My solution here was pretty much displaying a further error message to highlight the space, as was the most feasible

### CODE REVIEW

File: src/utils/document.js [hunk 1]
```diff
@@ -578,3 +578,11 @@ export const constructCSV = (results) => {
   });
   return stringify(tabularData);
 };
+
+export const getSpacedAttributeCode = (text) => {
+  if (text.includes('not a valid value for attribute')) {
+    const splitText = text.split('"');
+    const code = splitText[1];
```
Comment (akmiller01) [inline src/utils/document.js:585]: Good solution, Thatcher! I am wondering if there's a way to do this, to make sure that:

1. `splitText[1]` never throws an IndexError
2. All quoted strings in the context are considered
3. All whitespace is considered (e.g. spaces, linebreaks, non-breaking spaces)
4. Only whitespace at the beginning or the end of a string is considered, not all spaces inside the string

First, I would identify quoted sections of the string using regex, like:
```
const quotedStrings = text.match(/"(.*?)"/g).map(match => match.slice(1, -1));
```

And then we would be able to map the quoted strings to check whether the `trim()` function changes the string. Trim considers all possible types of whitespace, but only at the beginning or end of the string:
```
const noWhitespace = quotedStrings.map(str => str.trim() === str).every(bool => bool === true);
```

So all put together, I would redo the function to something like:
```
export const containsQuotedTrailingWhitespace = (text) => {
    const quotedStrings = text.match(/"(.*?)"/g).map(match => match.slice(1, -1));
    const noWhitespace = quotedStrings.map(str => str.trim() === str).every(bool => bool === true);
    return(!noWhitespace);
};
```

e.g.
```
> text
'This is an "example" of a "string" with multiple "quoted " sections.'
> containsQuotedTrailingWhitespace(text)
true
> text2
'This is an "example" of a "string" with multiple "quoted" sections.'
> containsQuotedTrailingWhitespace(text2)
false
```

I think that would be a little less brittle!

**Review by akmiller01 — CHANGES_REQUESTED** [review]
Good solution, Thatcher! Please see review for some ideas.

File: src/utils/document.js (hunk 1 above)
Comment (ThatcherK) [reply src/utils/document.js:585]: This does look robust thanks

**Review by akmiller01 — APPROVED** [review]
Looks good :+1:

**Pull request merged**